src/modules/peek/Peek.UI/Helpers/FileExplorerHelper.cs

```diff
@@ -2,27 +2,36 @@
 // The Microsoft Corporation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
-using System.Diagnostics;
-using Peek.Common.Models;
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
 using Peek.UI.Native;
+using Windows.Win32;
+using Windows.Win32.Foundation;
 
 namespace Peek.UI.Helpers
 {
     public static class FileExplorerHelper
     {
-        public static Shell32.IShellFolderViewDual2? GetCurrentFolderView()
+        public static unsafe Shell32.IShellFolderViewDual2? GetCurrentFolderView()
         {
             var foregroundWindowHandle = NativeMethods.GetForegroundWindow();
 
+            int capacity = PInvoke.GetWindowTextLength(new HWND(foregroundWindowHandle)) * 2;
+            StringBuilder foregroundWindowTitleBuffer = new StringBuilder(capacity);
+            NativeMethods.GetWindowText(new HWND(foregroundWindowHandle), foregroundWindowTitleBuffer, foregroundWindowTitleBuffer.Capacity);
+
+            string foregroundWindowTitle = foregroundWindowTitleBuffer.ToString();
+
             var shell = new Shell32.Shell();
             foreach (SHDocVw.InternetExplorer window in shell.Windows())
             {
-                // TODO: figure out which window is the active explorer tab
-                // https://github.com/microsoft/PowerToys/issues/22507
-                if (window.HWND == (int)foregroundWindowHandle)
+                var shellFolderView = (Shell32.IShellFolderViewDual2)window.Document;
+                var folderTitle = shellFolderView.Folder.Title;
+
+                if (window.HWND == (int)foregroundWindowHandle && folderTitle == foregroundWindowTitle)
```

Comment:
Curious, what happens if the folder names are the same?

Reply:
That's what Esteban commented as well. Doesn't work in this case...